File: dart/nth-prime/lib/nth_prime.dart
import 'dart:math';

bool isPrime(int n) {
  for (int i = 2; i <= sqrt(n).round(); i++) {
    if (n % i == 0) {
      return false;
    }
  }
  return true;
}

class NthPrime {
  int prime(int n) {
    if (n == 0) {
      throw ArgumentError("There is no zeroth prime");
    }
    List<int> primes = [2];
    int testVal = 3;
    while (primes.length < n) {
      if (isPrime(testVal)) {
        primes.add(testVal);
      }
      testVal++;
    }
    return primes[primes.length - 1];
  }
}

File: kotlin/nth-prime/src/main/kotlin/NthPrime.kt
object Prime {

    fun nth(n: Int): Int {
        if (n == 0) {
            throw IllegalArgumentException("There is no zeroth prime.")
        }
        var primes = mutableListOf<Int>()
        var test = 2
        while (primes.size < n) {
            if (isPrime(test)) {
                primes.add(test)
            }
            test++
        }
        return primes[n - 1]
    }

    fun isPrime(n: Int): Boolean {
        var test = 2
        while (test <= Math.sqrt(n.toDouble())) {
            if (n % test++ == 0) {
                return false
            }
        }
        return true
    }
}
